lib/Transform/LinalgToDsePass.cpp

```diff
@@ -245,62 +243,61 @@ struct LinalgToDsePass
+
+
+      if (auto conv = dyn_cast<linalg::Conv2DNchwFchwOp>(op)) {
+
         auto *outOp = conv.outputs()[0].getDefiningOp();
-        if (biases.count(outOp)) {
-          // We know that this is the bias copy for this conv2d.
-          // No further information is needed.
-          biases.remove(outOp);
-        }
+        biases.remove(outOp);
         auto *inOp = op->getOperand(0).getDefiningOp();
         auto name = nextName.getNext();
         OutputOp output("Conv2D");
-        if (pads.count(inOp)) {
+
+        mlir::RankedTensorType c2d_inDim = nullptr;
+        std::vector<int64_t> c2d_padDim;
+
+        if (pads.remove(inOp)) {
           auto *padOp = inOp;
-          pads.remove(padOp);
           inOp = padOp->getOperand(0).getDefiningOp();
           auto pad = dyn_cast<linalg::PadTensorOp>(padOp);
-          output.addConv2dPart(
-              /*name=*/conv->getAttr("layer_name"),
-              /*inDim=*/as_nhwc_array(pad.getSourceType()),
-              /*filterDim=*/
-              wgts_to_kernel_array(conv.inputs()[1]
-                                       .getType()
-                                       .dyn_cast<mlir::RankedTensorType>()),
-              /*strideDim=*/as_2d_array(conv.strides()),
-              /*padDim=*/as_array(pad.getMixedHighPad(), pad.getMixedLowPad()));
+
+          c2d_inDim = pad.getSourceType();
+          c2d_padDim = as_array(pad.getMixedHighPad(), pad.getMixedLowPad());
         } else {
-          auto conv = dyn_cast<linalg::Conv2DNchwFchwOp>(op);
-          output.addConv2dPart(
-              /*name=*/conv->getAttr("layer_name"),
-              /*inDim=*/
-              as_nhwc_array(conv.inputs()[0]
-                                .getType()
-                                .dyn_cast<mlir::RankedTensorType>()),
-              /*filterDim=*/
-              wgts_to_kernel_array(conv.inputs()[1]
-                                       .getType()
-                                       .dyn_cast<mlir::RankedTensorType>()),
-              /*strideDim=*/as_2d_array(conv.strides()),
-              /*padDim=*/{0, 0, 0, 0});
+          c2d_inDim =
+              conv.inputs()[0].getType().dyn_cast<mlir::RankedTensorType>();
+          c2d_padDim = {0, 0, 0, 0};
         }
+
+        auto c2d_name = conv->getAttr("layer_name");
+        auto c2d_filterDim =
+            conv.inputs()[1].getType().dyn_cast<mlir::RankedTensorType>();
+
+        output.addConv2dPart(
+            c2d_name,
+            as_nhwc_array(c2d_inDim),
+            wgts_to_kernel_array(c2d_filterDim),
+            as_2d_array(conv.strides()),
+            c2d_padDim);
+
         inputToResult[inOp] = {op->getResult(0), name};
         nodes.insert({name, std::move(output.object)});
 
         return WalkResult::skip();
       }
-      if (patternConvLreluMaxpool.match(op)) {
+
+
+      if (auto conv = dyn_cast<linalg::Conv2DLreluMaxpoolOp>(op))) {
+
         auto *inOp = op->getOperand(0).getDefiningOp();
```

Comment:
NIT: It would have been good to pull the code working on `conv` into a separate function.